tests/common.py

```diff
@@ -977,9 +977,6 @@ async def mock_psc(hass, config_input, integration):
     ), f"setup_component failed, expected {count} got {res_len}: {res}"
 
 
-SetupRecorderInstanceT = Callable[..., Awaitable[Any]]
-
-
```

Comment:
This was replaced by `from tests.typing import RecorderInstanceGenerator`
The last remaining references are included in this PR, so I took the opportunity to clean it up.
cc @emontnemery